pkg/kapp/diffgraph/change.go

```diff
@@ -152,6 +157,10 @@ func (c *Change) AllRules() ([]ChangeRule, error) {
 
 		if (ctlres.AnyMatcher{rms}).Matches(res) {
 			for _, ruleStr := range ruleConfig.Rules {
+				ruleStr, err := parseChangeGroupPlaceholders(c.Change.Resource(), ruleStr)
```

Comment:
parseChangeGroupPlaceholders will never return error as per implementation. If that function is not supposed to return error, then we can omit it.

Reply:
It's returning error in cases of CRDs, but we have to refactor this anyways, so we can discuss about it.

Reply:
I think what Rohit was trying to say is that when `crd.Name()` or `crd.Group()` errors out we return `parsed, nil` when we should be returning the error itself (`parsed, err`) in lines 251 and 255. I will go ahead and make this change

Reply:
Resolved in amended commit